lib/absinthe/subscription/supervisor.ex

```diff
@@ -24,22 +24,43 @@ defmodule Absinthe.Subscription.Supervisor do
     pool_size = Keyword.get(opts, :pool_size, System.schedulers_online() * 2)
     compress_registry? = Keyword.get(opts, :compress_registry?, true)
 
-    Supervisor.start_link(__MODULE__, {pubsub, pool_size, compress_registry?})
+    document_storage =
+      Keyword.get(opts, :document_storage, Absinthe.Subscription.DefaultDocumentStorage)
+
+    storage_opts =
+      case document_storage do
+        Absinthe.Subscription.DefaultDocumentStorage ->
+          [
+            keys: :duplicate,
+            partitions: System.schedulers_online(),
+            compressed: compress_registry?
+          ]
+
+        _ ->
+          Keyword.get(opts, :storage_opts, Keyword.new())
+      end
+
+    Supervisor.start_link(
+      __MODULE__,
+      {pubsub, pool_size, document_storage, storage_opts}
+    )
   end
 
-  def init({pubsub, pool_size, compress_registry?}) do
+  def init({pubsub, pool_size, document_storage, storage_opts}) do
     registry_name = Absinthe.Subscription.registry_name(pubsub)
-    meta = [pool_size: pool_size]
+    meta = [pool_size: pool_size, document_storage: document_storage]
+
+    storage_opts =
+      Keyword.put(storage_opts, :name, Absinthe.Subscription.document_storage_name(pubsub))
 
     children = [
       {Registry,
        [
-         keys: :duplicate,
+         keys: :unique,
          name: registry_name,
-         partitions: System.schedulers_online(),
-         meta: meta,
-         compressed: compress_registry?
+         meta: meta
```

Comment:
I didn't know where to keep the metadata so I kept a registry process around the just holds metadata

Reply:
I'm not sure what the answer is at this point either.